gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskInfo.java

```diff
@@ -0,0 +1,439 @@
+      this.diskId = checkNotNull(diskId);
+      return this;
+    }
+
+    @Override
+    BuilderImpl creationTimestamp(Long creationTimestamp) {
+      this.creationTimestamp = creationTimestamp;
+      return this;
+    }
+
+    @Override
+    BuilderImpl creationStatus(CreationStatus creationStatus) {
+      this.creationStatus = creationStatus;
+      return this;
+    }
+
+    @Override
+    public BuilderImpl description(String description) {
+      this.description = description;
+      return this;
+    }
+
+    @Override
+    BuilderImpl licenses(List<LicenseId> licenses) {
+      this.licenses = licenses != null ? ImmutableList.copyOf(licenses) : null;
+      return this;
+    }
+
+    @Override
+    BuilderImpl attachedInstances(List<InstanceId> attachedInstances) {
+      this.attachedInstances =
+          attachedInstances != null ? ImmutableList.copyOf(attachedInstances) : null;
+      return this;
+    }
+
+    @Override
+    BuilderImpl lastAttachTimestamp(Long lastAttachTimestamp) {
+      this.lastAttachTimestamp = lastAttachTimestamp;
+      return this;
+    }
+
+    @Override
+    BuilderImpl lastDetachTimestamp(Long lastDetachTimestamp) {
+      this.lastDetachTimestamp = lastDetachTimestamp;
+      return this;
+    }
+
+    @Override
+    public DiskInfo build() {
+      return new DiskInfo(this);
+    }
+  }
+
+  DiskInfo(BuilderImpl builder) {
+    this.id = builder.id;
+    this.configuration = checkNotNull(builder.configuration);
+    this.creationTimestamp = builder.creationTimestamp;
+    this.creationStatus = builder.creationStatus;
+    this.diskId = checkNotNull(builder.diskId);
+    this.description = builder.description;
+    this.licenses = builder.licenses;
+    this.attachedInstances = builder.attachedInstances;
+    this.lastAttachTimestamp = builder.lastAttachTimestamp;
+    this.lastDetachTimestamp = builder.lastDetachTimestamp;
+  }
+
+  /**
+   * Returns the creation timestamp in milliseconds since epoch.
+   */
+  public Long creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns the unique identifier for the disk; defined by the service.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the disk configuration.
+   */
+  @SuppressWarnings("unchecked")
+  public <T extends DiskConfiguration> T configuration() {
+    return (T) configuration;
+  }
+
+  /**
+   * Returns the disk identity.
+   */
+  public DiskId diskId() {
+    return diskId;
+  }
+
+  /**
+   * Returns the creation status of the disk.
+   */
+  public CreationStatus creationStatus() {
+    return creationStatus;
+  }
+
+  /**
+   * Returns a textual description of the disk.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns all applicable publicly visible licenses for the disk.
+   */
+  public List<LicenseId> licenses() {
+    return licenses;
+  }
+
+  /**
+   * Returns all instances' identities this disk is attached to.
```

Comment:
Nit: I think this could sound slightly clearer as "Returns all the identities of the instances this disk is attached to"